test/unit/state/siteUtilTest.js

```diff
@@ -425,9 +433,47 @@ describe('siteUtil', function () {
         const processedSites = siteUtil.addSite(Immutable.fromJS(sites), newSiteDetail, siteTags.BOOKMARK, oldSiteDetail)
         const expectedSiteKey = siteUtil.getSiteKey(expectedSiteDetail)
         let expectedSites = {}
-        expectedSites[expectedSiteKey] = expectedSiteDetail.toJS()
+        expectedSites[expectedSiteKey] = expectedSiteDetail.set('objectId', undefined).toJS()
         assert.deepEqual(processedSites.toJS(), expectedSites)
       })
+      it('sets an objectId when syncCallback is provided', function () {
```

Comment:
This appears to be describing and update/edit case if I am not mistaken. 

What I did not notice is a deletion test case.